packages/client/tests/functional/extensions/result.ts

```diff
@@ -380,4 +380,34 @@ testMatrix.setupTestSuite(() => {
       expectTypeOf(data).toEqualTypeOf<User>()
     }
   })
+
+  test('when args have both include and select and one of them is optional, result includes both', () => {
```

Comment:
This is probably incorrect anyways if we would think on a strict type-level, but the expected current behavior that was broken.